src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -189,7 +236,28 @@ public override string ToString()
             {
                 if (_cachedToString == null)
                 {
-                    _cachedToString = $"Request finished in {_elapsed.TotalMilliseconds}ms {_httpContext.Response.StatusCode} {_httpContext.Response.ContentType}";
+                    if (_logLevel <= LogLevel.Debug)
```

Comment:
Correlation isn't a concern assuming it's done inside the request is scope.

 Make it a separate event, Debug or possibly Verbose, also have one line per header with {HeaderName} and {HeaderValue} as structured message fields. Be sure the loop is inside an IsEnabled check.

That'll help when doing log analysis queries for things like distinct session id counts, user agent frequency, etc.